File: qc-core/src/main/java/com/eager/questioncloud/core/domain/cart/model/CartItem.java
package com.eager.questioncloud.core.domain.cart.model;

import lombok.Builder;
import lombok.Getter;

@Getter
public class CartItem {
    private Long id;
    private Long userId;
    private Long questionId;

    @Builder
    public CartItem(Long id, Long userId, Long questionId) {
        this.id = id;
        this.userId = userId;
        this.questionId = questionId;
    }

    public static CartItem create(Long userId, Long questionId) {
        return CartItem.builder()
            .userId(userId)
            .questionId(questionId)
            .build();
    }
}

File: qc-core/src/main/java/com/eager/questioncloud/core/domain/cart/model/CartItem.kt
package com.eager.questioncloud.core.domain.cart.model

class CartItem(
    val id: Long = 0,
    val userId: Long,
    val questionId: Long
) {
    companion object {
        fun create(userId: Long, questionId: Long): CartItem {
            return CartItem(userId = userId, questionId = questionId)
        }
    }
}

